Card Checking + Bug Fixes

Viomi opened the pull request:
Check if modem and/or tunnel, check isWireless(), handler fix, couple of typoes fixed, minor bug fixes, commented out undefined handleEvent()...

Commit:
Check if modem and/or tunnel, check isWireless(), handler fix, couple of typoes fixed, minor bug fixes, commented out undefined handleEvent()

Commit:
Better tunnel vs. modem checks

Commit:
Commented out another undefined handleEvent(), another error message for modem vs. tunnel usage.

MajorGeneralRelativity commented on GERTi/GERTi Client.lua:
```diff
@@ -34,7 +48,7 @@ local function storeNeighbors(eventName, receivingModem, sendingModem, port, dis
     neighbors[neighborDex]["port"] = tonumber(port)
     if package == nil then
-        --This is used for when a computer receives a new client's GERTiStart message. It stores a neighbor connection with a tier one lower than this computer's tier
+        --This is used for when a computer receives a new client's GERTiStart message. It stores a neighbor connection with a tier one higher than this computer's tier
```
It should be "a tier one lower", because a higher number indicates a client further away from the gateway, which is a lower tier.

MajorGeneralRelativity commented on GERTi/GERTi Client.lua:
```diff
@@ -162,22 +180,26 @@ handler."OPENROUTE" = local function RouteHandler(eventName, receivingModem, sen
         end
     end
+
+    return true
```
OpenRoute is always going to be returning something else, so the return true is unnecessary.

MajorGeneralRelativity commented on GERTi/GERTi Client.lua:
```diff
@@ -186,30 +208,30 @@ end
-end
+--if (tunnel) then
+--    tunnel.send("GERTiStart")
+--    handleEvent(event.pull(1, "modem_message"))
```
Please uncomment this block and replace handleEvent(event.pull(1, "modem_message")) with receivePacket(event.pull(1, "modem_message"))

MajorGeneralRelativity commented on GERTi/GERTi Client.lua:
```diff
@@ -186,30 +208,30 @@ end
+--while continue do
+--    continue = handleEvent(event.pull(2, "modem_message"))
+--end
+
```
Please uncomment this block and replace handleEvent(event.pull(2, "modem_message")) with receivePacket(event.pull(1, "modem_message"))

MajorGeneralRelativity commented on GERTi/GERTi Gateway.lua:
```diff
@@ -8,15 +8,24 @@ local childNum = 1
-modem.open(4378)
+
+if (component.isAvailable("modem")) then
+    local modem = component.modem
```
If a local is declared in this if...then block, will the modem variable be accessible outside? Same for the tunnel statement immediately below.

MajorGeneralRelativity reviewed (changes requested):
Please see the comments for a more detailed review. Looks good overall, but some tweaks should be made.

Commit:
Requested Changes Made

Changes previously made to Client now synchronized to Gateway

require tunnel and modem

Viomi commented:
See https://github.com/GlobalEmpire/GERT/pull/9/commits/a9835e4104dd7974fdb00b6f0116f2ae417a31aa - Made requested changes and a couple more.

MajorGeneralRelativity reviewed (approved):
Looks Good!
